Description: Only fix up powerpc key repeat on IEEE1275 machines.
 Fixes powerpc-emu compilation.
Author: Colin Watson <cjwatson@ubuntu.com>
Origin: upstream, http://bazaar.launchpad.net/~vcs-imports/grub/grub2-bzr/revision/4580
Forwarded: not-needed
Applied-Upstream: http://bazaar.launchpad.net/~vcs-imports/grub/grub2-bzr/revision/4580
Last-Update: 2012-09-14

Index: b/grub-core/term/terminfo.c
===================================================================
--- a/grub-core/term/terminfo.c
+++ b/grub-core/term/terminfo.c
@@ -33,7 +33,7 @@
 #include <grub/extcmd.h>
 #include <grub/i18n.h>
 #include <grub/time.h>
-#ifdef __powerpc__
+#if defined(__powerpc__) && defined(GRUB_MACHINE_IEEE1275)
 #include <grub/ieee1275/ieee1275.h>
 #endif
 
@@ -563,7 +563,7 @@
   grub_terminfo_readkey (termi, data->input_buf,
 			 &data->npending, data->readkey);
 
-#ifdef __powerpc__
+#if defined(__powerpc__) && defined(GRUB_MACHINE_IEEE1275)
   if (data->npending == 1 && data->input_buf[0] == '\e'
       && grub_ieee1275_test_flag (GRUB_IEEE1275_FLAG_BROKEN_REPEAT)
       && grub_get_time_ms () - data->last_key_time < 1000
@@ -580,7 +580,7 @@
       int ret;
       data->npending--;
       ret = data->input_buf[0];
-#ifdef __powerpc__
+#if defined(__powerpc__) && defined(GRUB_MACHINE_IEEE1275)
       if (grub_ieee1275_test_flag (GRUB_IEEE1275_FLAG_BROKEN_REPEAT))
 	{
 	  data->last_key = ret;
Index: b/include/grub/terminfo.h
===================================================================
--- a/include/grub/terminfo.h
+++ b/include/grub/terminfo.h
@@ -32,7 +32,7 @@
 {
   int input_buf[GRUB_TERMINFO_READKEY_MAX_LEN];
   int npending;
-#ifdef __powerpc__
+#if defined(__powerpc__) && defined(GRUB_MACHINE_IEEE1275)
   int last_key;
   grub_uint64_t last_key_time;
 #endif
